go/types/validating_decoder.go

```diff
@@ -14,9 +14,7 @@ type ValidatingDecoder struct {
 }
 
 func NewValidatingDecoder(cs chunks.ChunkStore) *ValidatingDecoder {
-	return &ValidatingDecoder{
-		newLocalValueStore(cs),
-	}
+	return &ValidatingDecoder{NewValueStore(cs)}
```

Comment:
...Isn't that just re-creating `ValidatingBatchingSink`? Minus the batching part, but I don't think that was really the piece that bothered you so.

Where do you see the server decoding chunks multiple times? I'm pretty sure they're decoded just once.